commands/deploy_command.go

```diff
@@ -330,14 +331,32 @@ func parseMtaArchiveArgument(rawMtaArchive interface{}) (bool, string) {
 }
 
 func (c *DeployCommand) uploadFromUrl(url string, mtaClient mtaclient.MtaClientOperations, namespace string,
-	disableProgressBar bool) (fileId, mtaId string, failure bool) {
-	progressBar := c.tryFetchMtarSize(url, disableProgressBar)
-
+	disableProgressBar bool) UploadFromUrlStatus {
 	encodedFileUrl := base64.URLEncoding.EncodeToString([]byte(url))
+
+	var uploadFromUrlStatus UploadFromUrlStatus
+	for i := 0; i < 3; i++ {
+		progressBar := c.tryFetchMtarSize(url, disableProgressBar)
+		uploadFromUrlStatus = c.doUploadFromUrl(encodedFileUrl, mtaClient, namespace, progressBar)
+		if shouldRetryUpload(uploadFromUrlStatus) {
+			ui.Warn("Retrying failed upload from URL operation")
+			continue
+		}
+		return uploadFromUrlStatus
+	}
```

Comment:
Can't we use some general retry mechanism? This custom retry logic is not configurable via `--retries` command flag.

Reply:
Refactored it a bit, so it could be reused. We cannot reuse the other retries, as one is bound to the rest client and the other retries the MTA operation.
Do you think it should be bound to the --retries, imo it should be used only for retrying the operation, not the upload.